source/PatchTopology.lagda

```diff
@@ -0,0 +1,886 @@
+      † = scott-continuous-join-eq (𝒪 X) (𝒪 X) j ζj S δ
+
+      ‡ : k (⋁[ 𝒪 X ] S) ≡ ⋁[ 𝒪 X ] ⁅ k s ∣ s ε S ⁆
+      ‡ = scott-continuous-join-eq (𝒪 X) (𝒪 X) k ζk S δ
+
+      ※ : ((⋁⟨ i ∶ I ⟩ j (S [ i ]) ∧[ 𝒪 X ] k (S [ i ]))
+            is-an-upper-bound-of
+           ⁅ 𝒮 m n ∣ (m , n) ∶ I × I ⁆) holds
+      ※ (m , n) = ∥∥-rec (holds-is-prop P) ε (pr₂ δ m n)
+       where
+        P : Ω 𝓥
+        P = 𝒮 m n
+             ≤[ poset-of (𝒪 X) ]
+            (⋁⟨ i ∶ I ⟩ j (S [ i ]) ∧[ 𝒪 X ] k (S [ i ]))
+
+        ε : Σ i ꞉ I , ((S [ m ]) ≤[ poset-of (𝒪 X) ] (S [ i ])
+                    ∧ ((S [ n ]) ≤[ poset-of (𝒪 X) ] (S [ i ]))) holds
+          → P holds
+        ε (i , p , q) =
+         𝒮 m n                                        ≡⟨ refl ⟩ₚ
+         j (S [ m ]) ∧[ 𝒪 X ] k (S [ n ])             ≤⟨ ♢    ⟩
+         j (S [ i ]) ∧[ 𝒪 X ] k (S [ n ])             ≤⟨ ♥    ⟩
+         j (S [ i ]) ∧[ 𝒪 X ] k (S [ i ])             ≤⟨ ♠    ⟩
+         ⋁⟨ i ∶ I ⟩ j (S [ i ]) ∧[ 𝒪 X ] k (S [ i ])  ■
+          where
+           ♢ = ∧[ 𝒪 X ]-left-monotone  (μj (S [ m ] , S [ i ]) p)
+           ♥ = ∧[ 𝒪 X ]-right-monotone (μk (S [ n ] , S [ i ]) q)
+           ♠ = ⋁[ 𝒪 X ]-upper ⁅ (j (S [ i ])) ∧[ 𝒪 X ] (k (S [ i ])) ∣ i ∶ I ⁆ i
+
+      i   = ∧[ 𝒪 X ]-left-monotone  (reflexivity+ (poset-of (𝒪 X)) †)
+      ii  = ∧[ 𝒪 X ]-right-monotone (reflexivity+ (poset-of (𝒪 X)) ‡)
+
+      iii = distributivity+ (𝒪 X) ⁅ j s ∣ s ε S ⁆ ⁅ k s ∣ s ε S ⁆
+
+
+      iv  = ⋁[ 𝒪 X ]-least
+             ⁅ 𝒮 m n ∣ (m , n) ∶ I × I ⁆
+             ((⋁⟨ i ∶ I ⟩ j (S [ i ]) ∧[ 𝒪 X ] k (S [ i ])) , ※)
+
+      v   = ⋁[ 𝒪 X ]-least ⁅ j (S [ i ]) ∧[ 𝒪 X ] k (S [ i ]) ∣ i ∶ I ⁆ 𝓊
+
+ _⋏_ : perfect-nucleus → perfect-nucleus → perfect-nucleus
+ (𝒿 , νj , ζj) ⋏ (𝓀 , νk , ζk) = pr₁ Σ-assoc (((𝒿 , νj) ⋏₁ (𝓀 , νk)) , γ)
+  where
+   μj = nuclei-are-monotone (𝒪 X) (𝒿 , νj)
+   μk = nuclei-are-monotone (𝒪 X) (𝓀 , νk)
+
+   γ : is-perfect (𝒿 ⋏₀ 𝓀) holds
+   γ = ⋏₀-perfect 𝒿 𝓀 μj μk ζj ζk
+
+\end{code}
+
+\section{Construction of the join}
+
+The construction of the join is the nontrivial component of this development.
+Given a family `S ∶≡ { fᵢ : A → A | i ∶ I }` of endofunctions on some type `A`,
+and a list `i₀, …, iₙ` of indices (of type `I`), the function `sequence gives
+the composition of all `fᵢₙ ∘ ⋯ ∘ fᵢ₀`:
+
+\begin{code}
+
+ sequence : {A : 𝓤 ̇ } → (S : Fam 𝓦 (A → A)) → List (index S) → A → A
+ sequence S []       = id
+ sequence S (i ∷ is) = sequence S is ∘ S [ i ]
+
+\end{code}
+
+Using `sequence`, we define the following functio that will help us “directify”
+a given family:
+
+\begin{code}
+
+ 𝔡𝔦𝔯 : {A : 𝓤 ̇ } (S : Fam 𝓦 (A → A)) → Fam 𝓦 (A → A)
+ 𝔡𝔦𝔯 S = List (index S) , sequence S
+
+\end{code}
+
+The first lemma we prove about `𝔡𝔦𝔯` is the fact that, given a family
+
+```
+S ∶≡ { jᵢ : 𝒪 X → 𝒪 X ∣ i ∶ I }
+```
+
+of prenuclei, `sequence S is` is a prenuclei for any given list `is : List I` of
+indices.
+
+\begin{code}
+
+ 𝔡𝔦𝔯-prenuclear : (K : Fam 𝓦 (⟨ 𝒪 X ⟩ → ⟨ 𝒪 X ⟩))
```

Comment:
"𝔡𝔦𝔯-prenuclear" is a terrible name. :-)
What about "family-of-finite-compositions"?